File: iOS/ssabree/ssabree/Core/Network/DTO/HomeResponse.swift
import Foundation

struct HomeResponse: Codable {
    let dDays: [HomeDDayResponse]
    let teamThumbnail: HomeRecruitThumbResponse?
    let studyThumbnail: HomeRecruitThumbResponse?
    let campusMeals: [HomeCampusMealResponse]
    let boardsList: [HomeBoardThumbResponse]
    // 백엔드가 camelCase로 반환하므로 CodingKeys 불필요
}

struct HomeDDayResponse: Codable {
    let title: String
    let days: Int
}

struct HomeRecruitThumbResponse: Codable {
    let name: String?
    let count: Int
}

struct HomeBoardThumbResponse: Codable {
    let boardId: Int
    let name: String
    let recentPostTitle: String?
    // 백엔드가 camelCase로 반환하므로 CodingKeys 불필요
}

struct HomeCampusMealResponse: Codable {
    let campusId: Int
    let campusName: String
    let imageUrls: [String]
    // 백엔드가 camelCase로 반환하므로 CodingKeys 불필요
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/remote/model/HomeResponse.kt
package com.ssafy.ssabree.core.datasource.remote.model

import com.google.gson.annotations.SerializedName

data class HomeResponse(
    @SerializedName("dDays") val dDays: List<HomeDdayResponse>,
    @SerializedName("teamThumbnail") val teamThumbnail: RecruitThumbResponse?,
    @SerializedName("studyThumbnail") val studyThumbnail: RecruitThumbResponse?,
    @SerializedName("campusMeals") val campusMeals: List<CampusMealResponse>,
    @SerializedName("boardsList") val boardsList: List<BoardThumbResponse>
)

data class HomeDdayResponse(
    @SerializedName("title") val title: String,
    @SerializedName("days") val days: Int
)

data class RecruitThumbResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String?,
    @SerializedName("count") val count: Int
)

data class BoardThumbResponse(
    @SerializedName("boardId") val boardId: Long,
    @SerializedName("name") val name: String,
    @SerializedName("recentPostTitle") val recentPostTitle: String?
)

data class CampusMealResponse(
    @SerializedName("campusId") val campusId: Long,
    @SerializedName("campusName") val campusName: String,
    @SerializedName("imageUrls") val imageUrls: List<String>
)
